digest/Cargo.toml

```diff
@@ -20,6 +20,7 @@ block-buffer = { version = "=0.11.0-pre.5", optional = true }
 subtle = { version = "2.4", default-features = false, optional = true }
 blobby = { version = "0.3", optional = true }
 const-oid = { version = "=0.10.0-pre.2", optional = true }
+spki = { version = "=0.8.0-pre.0", optional = true }
```

Comment:
Ugh, this has a circular dependency: `spki` has an optional dependency on `sha2` for computing SPKI fingerprints.

Reply:
Ha, I missed this one.

Newtype then ? :D 
And we get the `AssociatedOid` and `AssociatedAlgorithmIdentifier` attached to those ? :D

Reply:
If we had newtypes, it would at least get rid of any coupling between `spki` and `digest`, though the problem would remain for `sha2`.

We could potentially split off an e.g. `spki-fingerprint` crate so `spki` doesn't depend on `sha2`. It is something of a niche feature.